### PR TITLE
Add event logging for shift and car related events

### LINKED ISSUE #60: Add logging to track shift events
**Description:**
Add logging helper and log multiple events.

**Acceptance criteria:**

- [x] Add a log helper which calls the log endpoint from the backend
- [x] Track workflow events documented here: https://github.com/TU-CP18/cp18-backend/blob/master/log-collector/event-types.org#vehicle-interaction-and-ride
- [ ] Track control events like light switches and air conditioning documented here: https://github.com/TU-CP18/cp18-backend/blob/master/log-collector/event-types.org#control-events

### PR DESCRIPTION
**Pull request by nschoen**
Add a logging helper which calls the log collector api.  The `NAV_ESTOP` event will be covered in #19. The `CONTROL_SET` event will be used in #50. Closes #60.

Thanks for reviewing!

### CODE REVIEW

File: components/MapRoute.js [hunk 1]
```diff
@@ -394,6 +403,7 @@ MapRoute.defaultProps = {
   showNavigationButton: true,
   isNavigation: false,
   initialFocus: 'gps',
+  trackNavigationEvent: false,
```
Comment (calvinschmidt) [inline components/MapRoute.js:406]: Won't we log by default? I would expect `true` here.

File: store/User.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ export default class User {
       await asyncSleep(1000);
       const userDetails = JSON.parse(await AsyncStorage.getItem(USER_DETAILS));
       if (userDetails.id_token) {
+        this.id = userDetails.id;
```
Comment (calvinschmidt) [inline store/User.js]: duplicate

**Review by calvinschmidt — APPROVED** [review]
Looking good! I just have two minor comments.

File: components/MapRoute.js (hunk 1 above)
Comment (nschoen) [reply components/MapRoute.js:406]: This is NextShiftMapScreen specific to track the event that the user is on its way to the interchange point. When the component is embedded somewhere else, it should be disabled by default, e.g. in RideScreen. Alternatively we could use a navigationStarted callback and do the logging in the NextShiftScreen?

**Pull request merged**